File: backend-nodejs/src/agent/agent.types.ts
/** Scope for agent analysis: OEM's suppliers, locations, and commodities */
export interface OemScope {
  oemId: string;
  oemName: string;
  supplierNames: string[];
  locations: string[];
  cities: string[];
  countries: string[];
  regions: string[];
  commodities: string[];
}

File: backend/app/services/agent_types.py
from typing import TypedDict


class OemScope(TypedDict):
    oemId: str
    oemName: str
    supplierNames: list[str]
    locations: list[str]
    cities: list[str]
    countries: list[str]
    regions: list[str]
    commodities: list[str]
